src/View.php

```diff
@@ -570,6 +570,9 @@ public function popDynamicContent(): void
      */
     public function beginPage(): void
     {
+        if ($this->getViewFile() === null) {
+            throw new \LogicException('View file cannot be null.');
```

Comment:
We're not setting the file ourselves so the exception is the result of calling this method when no file is being rendered, right?

How about `\RuntimeException('beginPage() should be called in views only')`?

Reply:
How about `\LogicException('beginPage  should be called in views context only')`?

`\LogicException`:
> Exception that represents error in the program logic. **This kind of exception should lead directly to a fix in your code.**

Reply:
Yes, that's fine.